devlib/module/cgroups.py

```diff
@@ -287,6 +287,11 @@ def get_tasks(self):
         logging.debug('Tasks: %s', task_ids)
         return map(int, task_ids)
 
+    # Used to insert fake cgroup attach events to know existing cgroup assignments
+    def trace_cgroup_tasks(self):
```

Comment:
Honestly I think this can be a wrapper on top of my patch. I do believe my patch is clean enough but all that's missing is a wrapper to take controller and group parameters. IMO the core dumping into trace logic should be as low as possible (infact right fully in the `CGroup` class since we're dumping tasks per `CGroup`). Also I tried implementing it the other way - make the `CGroup controller` or `CgroupModule` do the dumping into trace and it complicates things because `self.tasks_file` is available in `CGroup` class and can just be directly accessed and passed along to the `shutil`. So I still like my approach to this is clean.

If you still want to do this differently, I am Ok with that. But I would like it to be as simple as my approach. Maybe you can just add a wrapper to this patch in `CgroupModule` if your concern is the interface?